projects/or_scs/src/Mirza/OrgRegistry/Types.hs

```diff
@@ -291,29 +291,58 @@ data LocationResponse = LocationResponse
   , geoLocAddress :: Maybe Text
   } deriving (Show, Generic, Eq)
 instance ToSchema LocationResponse
-instance ToJSON LocationResponse
-instance FromJSON LocationResponse
+instance ToJSON LocationResponse where
+  toJSON (LocationResponse locId locGLN pfix geolocId geolocCoord geolocAddr) = object
+    [ "location_id" .= locId
+    , "epc" .= locGLN
+    , "company_prefix" .= pfix
+    , "geolocation_id" .= geolocId
+    , "geolocation_coords" .= geolocCoord
+    , "geolocation_address" .= geolocAddr
+    ]
 
+instance FromJSON LocationResponse where
+  parseJSON = withObject "LocationResponse" $ \o -> LocationResponse
+    <$> o .: "location_id"
+    <*> o .: "epc"
+    <*> o .: "company_prefix"
+    <*> o .: "geolocation_id"
+    <*> o .:? "geolocation_coords"
+    <*> o .: "geolocation_address"
 
 data OrgAndLocationResponse = OrgAndLocationResponse
   { orgResponse      :: OrgResponse
   , locationResponse :: LocationResponse
   } deriving (Show, Generic, Eq)
 instance ToSchema OrgAndLocationResponse
-instance ToJSON OrgAndLocationResponse
-instance FromJSON OrgAndLocationResponse
-
+instance ToJSON OrgAndLocationResponse where
+  toJSON (OrgAndLocationResponse org loc) = object
+    [ "org" .= org
+    , "loc" .= loc
+    ]
+instance FromJSON OrgAndLocationResponse where
+  parseJSON = withObject "OrgAndLocationResponse" $ \o -> OrgAndLocationResponse
+    <$> o .: "org"
+    <*> o .: "location"
 
 data KeyState
   = InEffect -- Can be used
   | Revoked -- Key passed the revocation time
   | Expired -- Key passed the expiration time
   deriving (Show, Eq, Read, Generic)
-$(deriveJSON defaultOptions ''KeyState)
+instance FromJSON KeyState where
+  parseJSON = withText "KeyState" $ \case
+    "in_effect" -> pure InEffect
+    "revoked" -> pure Revoked
+    "expired" -> pure Expired
+    _ -> fail "Invalid value for key_state. Valid values are in_effect, revoked and expired."
+instance ToJSON KeyState where
+  toJSON InEffect = String "in_effect"
+  toJSON Revoked  = String "revoked"
+  toJSON Expired  = String "expired"
 instance ToSchema KeyState
 instance ToParamSchema KeyState
 
-
```

Comment:
Pedantic, but I think its good to have 2 spaces here. I've been trying to add 2 spaces between functions/logical groups and use single spaces within, just to assist with grouping readability.

Reply:
Oh, apologies. I wasn't aware of that. I will fix that.